engine_test.go

```diff
@@ -206,6 +206,32 @@ func TestQueries(t *testing.T) {
 	}
 }
 
+func TestJoinsFields(t *testing.T) {
```

Comment:
Can you add the query to the `queries` slice instead of create a new test for this?